custom_components/adaptive_lighting/hass_utils.py

```diff
@@ -39,15 +42,18 @@ def setup_service_call_interceptor(
     existing_service = registered_services[domain][service]
 
     async def service_func_proxy(call: ServiceCall) -> None:
-        # Convert read-only data to writeable dictionary for modification by interceptor
-        data = dict(call.data)
-
-        # Call interceptor
-        await intercept_func(call, data)
+        try:
+            # Convert read-only data to writeable dictionary for modification by interceptor
+            data = dict(call.data)
 
-        # Convert data back to read-only
-        call.data = ReadOnlyDict(data)
+            # Call interceptor
+            await intercept_func(call, data)
 
+            # Convert data back to read-only
+            call.data = ReadOnlyDict(data)
+        except Exception as e:  # noqa: BLE001
+            # Blindly catch all exceptions to avoid breaking light.turn_on
+            _LOGGER.error("Error in service_func_proxy: %s", e)
```

Comment:
@protyposis I think it's a good idea to catch all exceptions to avoid breaking `light.turn_on` accidentally.

Reply:
This was actually by intention, since there's nothing really that can go wrong in here, and I didn't want to use a logger in a utility function. An error would most likely happen in the implemented `intercept_func`, and I think it's the responsibility of the handler implementation to handle its own errors appropriately (which I also left out for simplicity... but sure, doesn't hurt to do this :) )

Reply:
Thanks for your reply!

Right, that makes sense, the `intercept_func` _should_ handle it. However, since anything can go wrong apparently (like things I would never expect (e.g., `self.hass is None` issue)), I thought it might be safer to wrap the entire thing in a try-catch all. Especially since a couple of folks reported that their lights wouldn't turn on.

Once we have a better understanding of why the `self.hass is None` happens, I would be happy to remove it again.